pkg/swagger/generator.go

```diff
@@ -0,0 +1,140 @@
+package swagger
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+
+	internalapi "github.com/Azure/ARO-RP/pkg/api"
+	v20200430 "github.com/Azure/ARO-RP/pkg/api/v20200430"
+	"github.com/Azure/ARO-RP/pkg/api/v20210131preview"
+)
+
+type generator interface {
+	ExampleOpenShiftClusterPutParameter() interface{}
+	ExampleOpenShiftClusterPatchParameter() interface{}
+	ExampleOpenShiftClusterResponse() interface{}
+	ExampleOpenShiftClusterCredentialsResponse() interface{}
+	ExampleOpenShiftClusterAdminKubeconfigResponse() interface{}
+	ExampleOpenShiftClusterListResponse() interface{}
+	ExampleOperationListResponse() interface{}
+
+	Features() features
+}
+
+const apiv20200430Path = "github.com/Azure/ARO-RP/pkg/api/v20200430"
+const apiv20210131previewPath = "github.com/Azure/ARO-RP/pkg/api/v20210131preview"
+
+var _ generator = &apiv20210131preview{}
+var _ generator = &apiv20200430{}
+
+type features struct {
+	kubeconfig         bool
```

Comment:
Easier to read and easier to understand. As much as it is possible I think consistency is better if we keep same behavior (feature flags). Otherwise some of them will be "flags, some function checks". Unreadable and insiders knowledge.